src/DynamoCore/Engine/EngineController.cs

```diff
@@ -609,14 +609,20 @@ public CompilationServices(LibraryServices libraryServices)
             priorNames = libraryServices.GetPriorNames();
         }
 
+        [Obsolete("This method is deprecated and will be removed in Dynamo 3.0")]
         /// <summary>
         /// Pre-compiles Design script code in code block node.
         /// </summary>
         /// <param name="parseParams">Container for compilation related parameters</param>
         /// <returns>true if code compilation succeeds, false otherwise</returns>
         public bool PreCompileCodeBlock(ref ParseParam parseParams)
         {
-            return CompilerUtils.PreCompileCodeBlock(compilationCore, ref parseParams, priorNames);
+            return CompilerUtils.PreCompileCodeBlock(compilationCore, parseParams, priorNames);
+        }
+
+        internal bool PreCompileCodeBlock(ParseParam parseParams)
```

Comment:
could we copy the summary stuff over if still applicable, helps to understand the method imo.